draft-ietf-quic-invariants.md

```diff
@@ -252,8 +252,9 @@ Packets for the same QUIC connection might use different connection ID values.
 
 ## Version
 
-QUIC versions are identified with a 32-bit integer, encoded in network byte
-order.  Version 0 is reserved for version negotiation (see
+The Version field contains a 32-bit integer, encoded in network byte order.
```

Comment:
(If this is off-topic for this PR, please ignore.)

Is there a reason why we specify that the Version field is an integer, as opposed to a 32-bit value?

Reply:
Only so that we can use numbers to talk about values.  TLS picked up this crazy {3, 3} notation that annoys the hell out of me.  People read that to mean that there are two discrete things with different semantics, when there is just one value and one semantic.  You could always write out the hex (as people now do for connection IDs), but then that's just a number.

Reply:
I definitely agree we don't want something like the TLS notation.

The only operation we need to do on QUIC versions (as encoded on the wire) is to check for equality. We don't need ordering of them (to check if one is less than another), and we certainly don't need to be able to do arithmetic on them, hence we don't need to consider them as numbers. By saying it's a 32-bit value (instead of an integer), we avoid needing to specify that it's network byte order. When specifying a version, you can write it out in hex (e.g. 0xff00001d), but that can be treated as a sequence of bytes instead of an integer.

Reply:
Would s/integer, .../identifier/ work for others?

Reply:
I personally like `32-bit identifier`. `32-bit opaque identifier` might be even better, or could be overkill.

Reply:
I'm fine with identifier.